Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogBehavior.cs

```diff
@@ -0,0 +1,211 @@
+
+            var poll = new CachedPromoDialog
+            {
+                MessageId = message.Id,
+                CampaignId = campaignSummary.Id,
+            };
+
+            SetDialogCache(campaign.Campaign.Id, message.Id, poll);
+            await _promotionDialogRepository.CreateAsync(message.Id, campaign.Campaign.Id);
+        }
+
+        private async Task DeletePromoDialog(PromotionActionCreationData campaign)
+        {
+            if (!_memoryCache.TryGetValue(campaign.Campaign.Id, out CachedPromoDialog poll))
+                return;
+            var pollChannel = await GetPollChannel(_discordSocketClient.GetGuild(campaign.GuildId));
+
+            var message =  await pollChannel.GetMessageAsync(poll.MessageId);
+
+            if (!await _promotionDialogRepository.TryDeleteAsync(message.Id))
+                throw new InvalidOperationException("Dialog to be deleted does not exist");
+
+            await message.DeleteAsync();
+        }
+
+        private EmbedBuilder BuildPollEmbed(PromotionCampaignSummary campaign, int Approve, int Oppose, string error = null)
+        {
+                var boldRole = $"**{MentionUtils.MentionRole(campaign.TargetRole.Id)}**";
+                var user = _discordSocketClient.GetUser(campaign.Subject.Id);
+
+                ValueTask<Color> aviColor;
+
+                if ((user.GetAvatarUrl(size: 16) ?? user.GetDefaultAvatarUrl()) is { } avatarUrl)
+                {
+                    aviColor = _imageService.GetDominantColorAsync(new Uri(avatarUrl));
+                }
+                else
+                {
+                    aviColor = new ValueTask<Color>(Color.Teal);
+                }
+
+                var embed = new EmbedBuilder()
+                    .WithTitle($"#{campaign.Id}")
+                    .WithDescription($"**{campaign.Subject.Nickname}** Has been nominated for promotion to {boldRole} ")
+                    .WithColor(aviColor.Result)
+                    .WithUserAsAuthor(user)
+                    .AddField("The current vote total stands at: ", $" {Approve} {_approveEmoji} / {Oppose} {_disapproveEmoji}")
+                    .WithTimestamp(_utcNow)
+                    .WithFooter("Please react with the appropriate reaction to cast your vote!");
+
+                return error == null ? embed : embed.AddField("⚠️ Vote Error: ⚠️", error);
+        }
+
+        private async Task<ITextChannel> GetPollChannel(IGuild guild)
+        {
+            var getPollChannel = await _designatedChannelService
+                .GetDesignatedChannelsAsync(guild, DesignatedChannelType.PromotionInterface);
+            return getPollChannel.First() as ITextChannel;
+        }
+
+        private void SetDialogCache(long campaign, ulong message, CachedPromoDialog dialog)
+        {
+            _memoryCache.Set(campaign, dialog);
+            _memoryCache.Set(message, dialog);
+        }
+
+        private static bool ValidateVoteReaction(IEmote emote)
```

Comment:
I'd call this maybe `IsRelevantReaction()` or `IsRelevantEmoji()`. You could also replace this with like `.GetDesiredSentiment()` that returns `null` if it's not a relevant emoji, then use that value for both the check and the service call.